dash/backend/src/modules/falco/service/falco.service.ts

```diff
@@ -120,6 +125,126 @@ export class FalcoService {
+    }
+
+    async findFalcoSettingByClusterId(clusterId: number): Promise<FalcoSettingDto>{
+        return this.falcoDao.findFalcoSettingByClusterId(clusterId);
+    }
+
+    async getAllAdminsToMail(): Promise<any>{
+        return this.falcoDao.getAllAdminsToMail();
     }
+
+    async sendFalcoEmail(emailReceiver: string, clusterId: number, falcoId: number, falcoSeverity: string, falcoNamespace: string, falcoSignature: string, newFalcoLog: FalcoDto): Promise<any>{
+        const savedTime = this.email.send({
+            to: `${emailReceiver}`,
+            from: this.configService.get('email.default.sender'),
+            subject: `New ${falcoSeverity} Project Falco Alert in ${falcoNamespace}`,
+            template: 'falco-log-email',
+            context: {
+                falcoLog: instanceToPlain(newFalcoLog[0]),
+                falcoLogTags: instanceToPlain(newFalcoLog[0].raw["tags"]),
+                moreDetailsLink: `https://dev-m9sweeper.intelletive.com/private/clusters/${clusterId}/falco/more/${falcoId}/signature/${falcoSignature}`,
+            }
+        }).catch(e => {
+            console.log('Error sending falco email: ' + e);
+        });
+        return savedTime;
+    }
+
+    async addFalcoEmail(emailSentTime: number, clusterId: number, falcoSignature: string): Promise<any>{
+        return this.falcoDao.addFalcoEmail(emailSentTime, clusterId, falcoSignature);
+    }
+
+    async falcoEmailAlreadySent(clusterId: number, falcoSignature: string): Promise<any>{
+        return this.falcoDao.falcoEmailAlreadySent( clusterId, falcoSignature);
+    }
+
+    async sendFalcoEmailAlert(clusterId: number, newFalcoLog: FalcoDto) {
+        const falcoSetting = await this.findFalcoSettingByClusterId(clusterId);
+
+        if (!falcoSetting) {
+            return; // no settings means there's no email to send!
+        }
+
+        // get all admin email addresses
+        const allAdminEmail = await this.getAllAdminsToMail();
+        // To test email: const allAdminEmailArray = ['some_email_address'] and comment out forEach();
+        const allAdminEmailArray = [''];
+        allAdminEmail.forEach( element => allAdminEmailArray.push(element.email));
+
+        // Parse data from new falco log fields
+        const falcoId = newFalcoLog[0].id;
+        const falcoSignature = newFalcoLog[0].anomaly_signature;
+        const falcoNamespace = newFalcoLog[0].namespace;
+        const falcoSeverity = newFalcoLog[0].level;
+
+        // Parse data from falco settings json fields
+        const severityLevel = falcoSetting["severity_level"];
+        const anomalyFrequency = (falcoSetting['anomaly_notification_frequency'] * 24 * 60 * 60 * 1000);
+        const weekDay = falcoSetting['weekday'];
+        const emailList = falcoSetting['email_list'].split(',');
+        const cleanEmailList = [];
+        for (const ele of emailList) {
+            cleanEmailList.push(ele.trim());
```

Comment:
If it is empty, it should not push to the list. 

I would refactor to... 

        const cleanEmailList = emailList.map(e => e.trim()).filter(e => e !== '');